### ISSUE TITLE
examples that don't work (old versions of netstat / df)

### ISSUE THREAD

**Issue by gmolveau** [kellyjonbrazil/jc#177]
Hi, I encountered two real case scenarios where jc didn't work.

Those scenarios come from old versions of tools (like netstat and df).

Here's a simple one (netstat) with spaces in the headers and blank fields without dash : 

```
Test_header_1             Test header 2       Test_Header_3         Test Header_4           testheader5
some data with spaces                 7       0.0.0.0:*             ESTABLISHED             some_data
some_data                             7       :::*                  LISTEN
some_data                             7       0.0.0.0:*                                     some data with spaces
some_data                            12       1.1.10.0:*
```

and a more difficult one (df) where the indentation is broken : 

```
Filesystem    Type 1024-blocks      Used Available Capacity Mounted on
/dev/mapper/VolGroup00-LogVol00 ext3     6030784   1147932   4571556      21% /
proc          proc           0         0         0       -  /proc
sysfs        sysfs           0         0         0       -  /sys
```

**Comment by kellyjonbrazil** [comment]
Thanks for reporting this! If you could provide actual `netstat` output that would be helpful as I might be able to fix that.

I'll see if I can find a fix for the `df` issue, as well.

**Comment by gmolveau** [comment]
Here's an example for netstat : 

```
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd
udp        0      0 200.4.30.128:68         0.0.0.0:*                           903/NetworkManager
Active UNIX domain sockets (servers and established)
Proto RefCnt Flags       Type       State         I-Node PID/Program name    Path
unix  2      [ ACC ]     STREAM     LISTENING      18438 1/systemd           /run/lvm/lvmpolld.socket
unix  2      [ ]         DGRAM                     23569 871/chronyd         /var/run/chrony/chronyd.sock
unix  2      [ ACC ]     STREAM     LISTENING      23584 872/gssproxy        /run/gssproxy.sock
unix  2      [ ACC ]     STREAM     LISTENING      18511 1/systemd           /run/rpcbind.sock
unix  2      [ ACC ]     SEQPACKET  LISTENING      18521 1/systemd           /run/udev/control
unix  3      [ ]         DGRAM                     11098 1/systemd           /run/systemd/notify
unix  2      [ ]         DGRAM                     11100 1/systemd           /run/systemd/cgroups-agent
unix  2      [ ACC ]     STREAM     LISTENING      24236 863/sssd            /var/lib/sss/pipes/private/sbus-monitor
```

**Comment by kellyjonbrazil** [comment]
I think I found the issue - will need to play around with it a bit to make sure I don't break anything with the fix. I think it comes down to the fact that I removed the dash in `I-Node` when normalizing, which shifted the inode/program_name data to the right by one character.

**Comment by kellyjonbrazil** [comment]
`netstat` issue fixed in `dev`. https://github.com/kellyjonbrazil/jc/commit/83371edd8fe3d23dfa92fc707604086522c85ab9

Working on the `df` issue.

**Comment by kellyjonbrazil** [comment]
I have a fix for the `df` issue in `dev`. https://github.com/kellyjonbrazil/jc/commit/cd7731484d826d84d835ecd460d746e78cadc5fc

I basically look to see if the filesystem field overflows the column length and, if so, create a truncated hash value so it can be parsed. Then after parsing, I replace the truncated hash value with the original value.

You can test this by copy/pasting the code here: https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/df.py

into a file named `df.py` in your "App Data Directory". (See here: https://github.com/kellyjonbrazil/jc#custom-parsers)

Let me know if you find any issues.

Thanks!

**Comment by kellyjonbrazil** [comment]
I have added these fixes into v1.17.2 available via pip install. Let me know if you run into any other issues. Thanks!

**Comment by gmolveau** [comment]
Awesome ! Thank's a lot for your work :)
